Sources/Rendering/OpenGL/Convolution2DPass/index.js

```diff
@@ -0,0 +1,311 @@
+          'vertexDC',
+          model.tris.getCABO().getVertexOffset(),
+          model.tris.getCABO().getStride(),
+          gl.FLOAT,
+          3,
+          gl.FALSE
+        )
+      ) {
+        vtkErrorMacro('Error setting vertexDC in copy shader VAO.');
+      }
+      if (
+        !model.copyVAO.addAttributeArray(
+          program,
+          model.tris.getCABO(),
+          'tcoordTC',
+          model.tris.getCABO().getTCoordOffset(),
+          model.tris.getCABO().getStride(),
+          gl.FLOAT,
+          2,
+          gl.FALSE
+        )
+      ) {
+        vtkErrorMacro('Error setting vertexDC in copy shader VAO.');
+      }
+    } else {
+      viewNode.getShaderCache().readyShaderProgram(model.convolutionShader);
+    }
+
+    gl.viewport(0, 0, size[0], size[1]);
+    gl.scissor(0, 0, size[0], size[1]);
+
+    // activate texture
+    const tex = model.framebuffer.getColorTexture();
+    tex.activate();
+    model.convolutionShader.setUniformi('u_image', tex.getTextureUnit());
+    model.convolutionShader.setUniform2f(
+      'u_textureSize',
+      tex.getWidth(),
+      tex.getHeight()
+    );
+    model.convolutionShader.setUniformfv('u_kernel', model.kernel);
+    model.convolutionShader.setUniformf(
+      'u_kernelWeight',
+      publicAPI.computeKernelWeight(model.kernel)
+    );
+
+    // render quad
+    gl.drawArrays(gl.TRIANGLES, 0, model.tris.getCABO().getElementCount());
+    tex.deactivate();
+  };
+
+  publicAPI.getFragmentShaderCode = (kernelDimension) => {
+    // generate new shader code
+    const kernelLength = kernelDimension * kernelDimension;
+    let shaderCode = [
+      '//VTK::System::Dec',
+      '//VTK::Output::Dec',
+      'uniform sampler2D u_image;',
+      'uniform vec2 u_textureSize;',
+      `uniform float u_kernel[${kernelLength}];`,
+      'uniform float u_kernelWeight;',
+      'varying vec2 tcoord;',
+      'void main(){',
+      '    vec2 onePixel = vec2(1.0, 1.0) / u_textureSize;',
+      '    vec4 colorSum =\n',
+    ].join('\n');
+
+    const halfDim = Math.floor(kernelDimension / 2);
+
+    // generate sum per pixel
+    let i = 0;
+    for (let y = -halfDim; y <= halfDim; ++y) {
+      for (let x = -halfDim; x <= halfDim; ++x) {
+        shaderCode += `        texture2D(u_image, tcoord + onePixel * vec2(${x}, ${y})) * u_kernel[${i}]`;
+        ++i;
+
+        if (i !== kernelLength) {
+          shaderCode += ' +\n';
+        }
+      }
+    }
+
+    // finish code
+    shaderCode += [
+      ';',
+      '    gl_FragData[0] = vec4((colorSum / u_kernelWeight).rgb, texture2D(u_image, tcoord).a);',
+      '}',
+    ].join('\n');
+
+    return shaderCode;
+  };
+
+  // build vertices etc
+  publicAPI.buildVertexBuffer = () => {
+    // 4 coner points in clipping space in order (x, y, z) where z is always set to -1
+    const ptsArray = new Float32Array([
+      -1,
+      -1,
+      -1,
+      1,
+      -1,
+      -1,
+      -1,
+      1,
+      -1,
+      1,
+      1,
+      -1,
```

Comment:
you could have used 4 lines for readability

Reply:
To follow-up: if `npm run reformat` is resulting in values on each line, then put `// prettier-ignore` above line 226.